week3/movielens.Rmd

```diff
@@ -1,92 +1,194 @@
-# store the result in a new data frame so you can use it in creating figure 2 from the paper below
-
-# plot the CDF of movie popularity
-```
-
-
-# Per-user stats
-
-```{r aggregate-by-user}
-# aggregate ratings by user, computing mean and number of ratings
-```
-
-```{r dist-user-activity}
-# plot distribution of user activity (= number of ratings the user made)
-# hint: try a log scale here
-```
-
-# Anatomy of the long tail
-
-```{r long-tail}
-# generate the equivalent of figure 2 of this paper:
-# https://5harad.com/papers/long_tail.pdf
-
-# Specifically, for the subset of users who rated at least 10 movies,
-# produce a plot that shows the fraction of users satisfied (vertical
-# axis) as a function of inventory size (horizontal axis). We will
-# define "satisfied" as follows: an individual user is satisfied p% of
-# the time at inventory of size k if at least p% of the movies they
-# rated are contained in the top k most popular movies. As in the
-# paper, produce one curve for the 100% user satisfaction level and
-# another for 90%---do not, however, bother implementing the null
-# model (shown in the dashed lines).
-```
+---
+title: "Movielens"
+date: '`r Sys.time()`'
+output:
+  html_document:
+    #code_folding: hide
+    number_sections: yes
+    toc: yes
+    toc_depth: 3
+editor_options: 
+  chunk_output_type: inline
+---
+
+```{r setup, include=FALSE}
+library(scales)
+library(tidyverse)
+library(knitr)
+
+# set plot theme
+theme_set(theme_bw())
+```
+
+# Load and preview data
+
+Read data from the `ratings.csv` file
+```{r load-data}
+ratings <- read_csv('ratings.csv',
+                    col_names = c('user_id','movie_id','rating','timestamp'))
+```
+
+Loaded `r format(object.size(ratings), units="Mb")` of ratings data, containing `r format(nrow(ratings), big.mark = ",")` ratings. Here's a preview:
+```{r preview-data}
+head(ratings) %>% kable()
+```
+
+# Summary statistics
+
+```{r dist-ratings}
+# plot the distribution of rating values https://speakerdeck.com/jhofman/modeling-social-data-lecture-2-introduction-to-counting?slide=26
+ratings %>%
+  ggplot(aes(x = rating)) +
+  geom_bar() +
+  scale_x_continuous(breaks = pretty_breaks(n = 10))+
+  scale_y_continuous(label= comma)
+  
+```
+
+## Per-movie stats
+
+```{r aggregate-by-movie}
+# aggregate ratings by movie, computing mean and number of ratings
+# hint: use the n() function for easy counting within a group
+movie_ratings <- ratings %>%
+  group_by(movie_id) %>%
+  summarize(num_ratings = n(), avg_rating = mean(rating))
+
+movie_ratings %>% head
+
+```
+
+```{r dist-movie-popularity}
+# plot distribution of movie popularity (= number of ratings the movie received)
+# hint: try scale_x_log10() for a logarithmic x axis
+# movie_ratings %>%
+#   ggplot(aes(x = movie_id, y = num_ratings)) +
+#   geom_point() + 
+#   scale_x_log10(label = comma) +
+#   scale_y_log10()
+movie_ratings %>%
+  ggplot(aes(x = num_ratings)) + 
+  geom_histogram(bins = 80) + 

```

Comment:
Why 80 bins? Does histogram not automatically determine the number of bins? Again - avoid magic numbers like this, as if you come back to this code in the future, it may be difficult to remember why you selected it!